### PR TITLE
Trusted images

### PR DESCRIPTION
**Pull request by juhmelo**
Adding `mark-trusted` command https://github.com/zalando-stups/pierone-cli/issues/64

### CODE REVIEW

File: pierone/cli.py [hunk 1]
```diff
@@ -300,6 +300,90 @@ def scm_source(config, team, artifact, tag, url, output):
                     max_column_widths={'revision': 10})
 
 
+def query_yes_no(question, default="yes"):
```
Comment (hjacobs) [inline pierone/cli.py]: this function IMHO does not make sense, you can use one of the provided `click` functions: http://click.pocoo.org/5/prompts/

File: pierone/cli.py [hunk 2]
```diff
@@ -300,6 +300,90 @@ def scm_source(config, team, artifact, tag, url, output):
...
+
+        valid = row.get('valid')
+
+        if valid is not True:
+            raise click.ClickException('SCM source information is not valid, cannot mark as trusted.')
+        else:
+            if query_yes_no('Do you want to mark this image as trusted?'):
+                request_post(config.get('url'), '/teams/{}/artifacts/{}/tags/{}/approval'.format(team, artifact, tag),
+                             None, token, False)
+                print('\x1b[0;32m' + 'Marked image as trusted.' + '\x1b[0m')
```
Comment (hjacobs) [inline pierone/cli.py]: Please don't hardcode escape codes and use `click`'s features instead

File: pierone/api.py [hunk 3]
```diff
@@ -116,6 +116,19 @@ def request(url, path, access_token: str = None, not_found_is_none: bool = False
         return r
 
 
+def request_post(url, path, body, access_token: str = None, not_found_is_none: bool = False) -> requests.Response:
```
Comment (hjacobs) [inline pierone/api.py]: it would be better to extend the existing `request` function to allow specifying method and data

File: pierone/api.py [hunk 4]
```diff
@@ -116,6 +116,19 @@ def request(url, path, access_token: str = None, not_found_is_none: bool = False
...
+    if access_token:
+        headers = {'Authorization': 'Bearer {}'.format(access_token)}
+    else:
+        headers = {}
+    r = session.post('{}{}'.format(url, path), headers=headers, timeout=10)
```
Comment (hjacobs) [inline pierone/api.py]: there is no data passed to `post`, this does not make sense

File: pierone/api.py [hunk 5]
```diff
@@ -103,12 +103,18 @@ def docker_login_with_iid(url):
...
-def request(url, path, access_token: str = None, not_found_is_none: bool = False) -> requests.Response:
+def request(url, path, access_token: str = None,
+            not_found_is_none: bool = False, method: str = 'GET', data=None) -> requests.Response:
     if access_token:
         headers = {'Authorization': 'Bearer {}'.format(access_token)}
     else:
         headers = {}
-    r = session.get('{}{}'.format(url, path), headers=headers, timeout=10)
+    if method == 'GET':
+        r = session.get('{}{}'.format(url, path), headers=headers, timeout=10)
```
Comment (jmcs) [inline pierone/api.py]: You can use `session.request(method, ...`

**Comment by ethercrow** [comment]
:+1:

**Comment by juhmelo** [comment]
:+1:

File: pierone/cli.py [hunk 6]
```diff
@@ -300,6 +300,57 @@ def scm_source(config, team, artifact, tag, url, output):
...
+
+        valid = row.get('valid')
+
+        if not valid:
+            raise click.ClickException('SCM source information is not valid, cannot mark as trusted.')
+        else:
+            if click.confirm('Do you want to mark this image as trusted?'):
+                request(config.get('url'), '/teams/{}/artifacts/{}/tags/{}/approval'.format(team, artifact, tag),
+                        access_token=token, method='POST', data=None)
+                click.echo(click.style('Marked image as trusted.', fg='green'))
```
Comment (hjacobs) [inline pierone/cli.py]: you can use `secho` instead which allows to pass the style parameter http://click.pocoo.org/5/utils/

**Comment by jmcs** [comment]
:+1:

**Comment by hjacobs** [comment]
:+1:

**Comment by juhmelo** [comment]
:+1:

**Comment by juhmelo** [comment]
:+1:

**Pull request merged**